cmd/mtoy/mtoy_test.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/walterbio/mtoy"
+)
+
+func TestRevealAnswer(t *testing.T) {
+	want := mtoy.New(time.Now().Unix()).RevealAnswer() + "\n"
+	var out bytes.Buffer
+	revealAnswer(&out)
+	got := out.String()
+	if got != want {
+		t.Errorf("got: %s, want: %s", got, want)
+	}
+}
+
+func TestMainFailWhenNoArgs(t *testing.T) {
+	t.Skip("this not the way to test main()")
+	cmd := exec.Command(os.Args[0], "-test.run=TestMainFailWhenNoArgsXXYY")
+	t.Log(len(os.Args))
+	t.Log(os.Args)
+	err := cmd.Run()
+	e, ok := err.(*exec.ExitError)
+	if ok {
+		if code := e.ExitCode(); code != 1 {
+			t.Errorf("main should exit with ExitCode == 1, got: %d", code)
+		}
+
+	}
+	if !ok {
+		t.Error("main should fail when no arguments")
+	}
+}
```

Comment:
well, I did realize that I am not executing the main program, I am executing the test program :). So this is not the way to to test that the main function exit with a corresponding error.